src/components/cc-dialog-confirm-form/cc-dialog-confirm-form.js

```diff
@@ -0,0 +1,184 @@
+import { css, html, LitElement } from 'lit';
+import { createRef, ref } from 'lit/directives/ref.js';
+import { EventHandler } from '../../lib/events.js';
+import { formSubmit } from '../../lib/form/form-submit-directive.js';
+import { Validation } from '../../lib/form/validation.js';
+import { i18n } from '../../translations/translation.js';
+import { CcDialogConfirmEvent } from '../cc-dialog-confirm-actions/cc-dialog-confirm-actions.events.js';
+import '../cc-dialog-confirm-actions/cc-dialog-confirm-actions.js';
+import { CcDialog } from '../cc-dialog/cc-dialog.js';
+import '../cc-input-text/cc-input-text.js';
+
+/**
+ * @import { Validator } from '../../lib/form/validation.js';
+ * @import { Ref } from 'lit/directives/ref.js';
+ */
+
+/**
+ * A form component to be used inside a `<cc-dialog>` for confirmation actions that require text input verification.
+ *
+ * This component displays a text input field where users must type a specific confirmation text (defined by `confirmTextToInput`)
+ * to enable form submission. The form automatically resets when the parent `<cc-dialog>` closes.
+ *
+ * ## Features
+ * - Required text input with custom validation
+ * - Submit and cancel actions via `<cc-dialog-confirm-actions>`
+ * - Automatic form reset on dialog close
+ * - Waiting state that disables all inputs and shows loading indicator
+ * - Customizable error messages and button labels
+ *
+ * @cssdisplay block
+ */
+export class CcDialogConfirmForm extends LitElement {
+  static get properties() {
+    return {
+      autofocusInput: { type: Boolean, attribute: 'autofocus-input' },
+      cancelLabel: { type: String, attribute: 'cancel-label' },
+      confirmErrorMessage: { type: String, attribute: 'confirm-error-message' },
+      confirmInputLabel: { type: String, attribute: 'confirm-input-label' },
+      confirmTextToInput: { type: String, attribute: 'confirm-text-to-input' },
+      submitIntent: { type: String, attribute: 'submit-intent' },
+      submitLabel: { type: String, attribute: 'submit-label' },
+      waiting: { type: Boolean, reflect: true },
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @type {boolean} Automatically focuses the input when the dialog is opened.
```

Comment:
nitpick: I'd break a line here, moreover it makes the type not highlighted on some IDEs by being here